source/portable/NetworkInterface/Common/phyHandling.c

```diff
@@ -626,12 +627,12 @@ BaseType_t xPhyStartAutoNegotiation( EthernetPhy_t * pxPhyObject,
 
             if( ( ulRegValue & phyBMSR_LINK_STATUS ) != 0 )
             {
-                ulPHYLinkStatus |= phyBMSR_LINK_STATUS;
+                ulPHYLinkStatus = phyBMSR_LINK_STATUS;
                 pxPhyObject->ulLinkStatusMask |= ulBitMask;
             }
             else
```

Comment:
You can leave out the else clause because `ulPHYLinkStatus` is already cleared during its declaration.